packages/@aws-cdk/aws-servicecatalog/lib/private/product-stack-synthesizer.ts

```diff
@@ -16,7 +25,17 @@ export class ProductStackSynthesizer extends cdk.StackSynthesizer {
   }
 
   public addFileAsset(_asset: cdk.FileAssetSource): cdk.FileAssetLocation {
-    throw new Error('Service Catalog Product Stacks cannot use Assets');
+    if (!this.stack) {
+      throw new Error('You must call bindStack() first');
```

Comment:
this message will reach the user. Do you mean to tell your user to call `bindStack()`?

Reply:
This is the error we throw in our `synthesize` method as well. I'm not sure about this, I don't think this is an error the user can act on regardless. I don't think we can ever reach this state, so maybe this error message is just for our own sanity.